src/main/scala/com/fasterxml/jackson/module/scala/experimental/ScalaObjectMapper.scala

```diff
@@ -67,18 +67,18 @@ trait ScalaObjectMapper {
         throw new IllegalArgumentException("Need exactly 2 type parameters for map like types ("+clazz.getName+")")
       }
       getTypeFactory.constructMapLikeType(clazz, typeArguments(0), typeArguments(1))
-    } else if (isCollectionLike(clazz)) {
+    } else if (isReference(clazz)) { // Option is a subclss of IterableOnce, so check it first
```

Comment:
Here's the core of the fix: Option needs to be checked before IterableOnce, since Option does extend IterableOnce now, but it needs its own mapper.